**Predict1:** Now predict the home price of home whoes dimensions **area= 3000sqft, no of bedrooms= 3  and age=40** of home given.

**Predict2:** Now predict the home price of home whoes dimensions **area= 2500sqft, no of bedrooms= 4  and age=5** of home given

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn import linear_model
from sklearn.linear_model import LinearRegression

In [2]:
df= pd.read_csv("homeprices.csv")
df
#Dataframe is ready

area  bedrooms  age   price
0  2600       3.0   20  550000
1  3000       4.0   15  565000
2  3200       NaN   18  610000
3  3600       3.0   30  595000
4  4000       5.0    8  760000
5  4100       6.0    8  810000

## DATA Preprocessing step 

In [3]:
# i have to take the median of values of bedrooms column
import math
x= math.floor(df['bedrooms'].median())
x

4

#The floor Function in Python is used to return the closest integer value which is less than or equal to given numeric value.

Bedrooms should be whole number. So i have to import math

I can fill Nan values using **fillna()** function. It will be available in pandas series. Now i got series in nan values is repalced with median number

In [4]:
df['bedrooms'].fillna(x)


0    3.0
1    4.0
2    4.0
3    3.0
4    5.0
5    6.0
Name: bedrooms, dtype: float64

In [5]:
df['bedrooms']=df['bedrooms'].fillna(x)

In [6]:
df

area  bedrooms  age   price
0  2600       3.0   20  550000
1  3000       4.0   15  565000
2  3200       4.0   18  610000
3  3600       3.0   30  595000
4  4000       5.0    8  760000
5  4100       6.0    8  810000

Before apllying machine learnig model,you have to preprocess you data, you have to clean your data, data may be massy, fix the error  and prepare your data and then apply your machine learning model.

In [7]:
# applying machine learning model
#create an regression class object
reg=linear_model.LinearRegression()
reg.fit(df[["area","bedrooms","age"]],df.price)# fit is used for train the models

LinearRegression()

Now your model has trained with trainig data. Now your created model is ready for predictions

 When your model become ready then this is good idea to find your coeficients

In [8]:
reg.coef_

array([  112.06244194, 23388.88007794, -3231.71790863])

In [9]:
reg.intercept_

221323.00186540408

Predict1: Now predict the home price of home whoes dimensions area= 3000sqft, no of bedrooms= 3  and age=40 of home given

In [10]:
reg.predict([[3000,3,40]])

array([498408.25158031])

Now Predicted Price is=498408.25158031

Now we can check how it will calculate 

In [11]:
112.06244194*3000+23388.88007794*3+-3231.71790863*40+221323.00186540408

498408.2515740241

Now we have to see various factor that can affect

**Predict2:** Now predict the home price of home whoes dimensions area= 2500sqft, no of bedrooms= 4  and age=5 of home given

In [12]:
reg.predict([[2500,4,5]])

array([578876.03748933])